pkg/controller/utils/utils.go

```diff
@@ -79,6 +82,7 @@ func IgnoreObject(obj runtime.Object) bool {
 	return false
 }
 
+// TODO: Deprecate and delete these functions.
```

Comment:
At a minimum these need to be updated to include the particular handler that should be used, instead of assuming EnqueueRequestForObject.

So the new signatures will be:

```go
if err := AddInstallationWatch(c, &handler.EnqueueRequestForObject{})
```

instead of, for example:

```go
if err := c.Watch(&source.Kind{Type: &operatorv1.Installation{}), &handler.EnqueueRequestForObject{})
```

I guess there is a small readability difference, at the cost of it being tedious to make changes (especially per-controller changes) when needed. I added this TODO because the use of these particular "helpers" wasted time for me at some point throughout this changeset. 

I'll remove the TODO for now since we're not in agreement, but I'm in general pretty unhappy with the proliferation of functions that simply wrap another function. I've said it before in the context of the "GetSecret" and "GetConfigMap" functions that just hide certain errors as well.

Reply:
They're not even consistently implemented - there's only `AddNetworkWatch`, `AddComplianceWatch`, and `AddAPIServerWatch`

I definitely would not say there is value in going through the code and adding equivalent functions for every other thing that we watch.